### ISSUE TITLE
Bannered shields cause neighboring shields to move too far down

### ISSUE THREAD

**Issue by claysmalley** [ZeLonewolf/openstreetmap-americana#340]
With shields along concurrent routes where one shield has more banners than another, the shield with fewer banners displays farther down on the screen than expected:

![Screenshot from 2022-05-21 12-03-13](https://user-images.githubusercontent.com/1732117/169659866-52d2864b-8da2-414b-b141-a202b5d5558f.png)

@ZeLonewolf:

> [It would require a change to maplibre to adjust the baseline for shields, or some other pixel gymnastics that I haven't thought of yet.](https://github.com/ZeLonewolf/openstreetmap-americana/pull/337#issuecomment-1133661668)

**Comment by ZeLonewolf** [comment]
The issue is that shields are drawn with a common baseline for the entire string, and drawn vertically aligned to the bottom of the baseline, like so:

![image](https://user-images.githubusercontent.com/3254090/169660594-bef09893-dee4-4b8b-8d2d-a46bef847cc1.png)

**Comment by zekefarwell** [comment]
Yeah this is a pretty fundamental issue with the way we are handling banners.  The banners are part of the shield graphic and MapLibre aligns the shield graphics in a concurrency to the baseline because they are being used as glyphs in a text string.  I'm guessing a fix for this issue would require changes to MapLibre.

**Comment by 1ec5** [comment]
Each string or `image` expression in a [`format`](https://maplibre.org/maplibre-gl-js-docs/style-spec/expressions/#types-format) expression can be followed optionally by an style override options object. This would call for an option similar to the [`vertical-align`](https://developer.mozilla.org/en-US/docs/Web/CSS/vertical-align) CSS property.

**Comment by 1ec5** [comment]
> Each string or `image` expression in a [`format`](https://maplibre.org/maplibre-gl-js-docs/style-spec/expressions/#types-format) expression can be followed optionally by an style override options object. This would call for an option similar to the [`vertical-align`](https://developer.mozilla.org/en-US/docs/Web/CSS/vertical-align) CSS property.

This is now available as of maplibre/maplibre-gl-js#5043, which is in MapLibre GL JS v5.1.0.
